homeassistant/components/hardware/models.py

```diff
@@ -17,12 +17,24 @@ class BoardInfo:
     revision: str | None
 
 
+@dataclass
+class USBInfo:
+    """USB info type."""
+
+    vid: str
+    pid: str
+    serial_number: str | None
+    manufacturer: str | None
+    description: str | None
+
+
 @dataclass
```

Comment:
Don't forget to freeze
```suggestion
@dataclass(freeze=True)
```